BraveShared/Analytics/DAU.swift

```diff
@@ -199,47 +221,31 @@ public class DAU {
         if firstPing {
             return dauParams(true, true, true)
         }
-        
-        let month = todayComponents.month
-        let year = todayComponents.year
-        
+
         guard let stat = dauStat?.compactMap({ $0 }) else {
             log.error("Cannot cast dauStat to [Int]")
             return nil
         }
         
-        guard stat.count == 3 else {
-            log.error("dauStat array must contain exactly 3 elements")
+        guard let lastPingStat = stat.first else {
+            log.error("Can't get last ping timestamp from dauStats")
             return nil
         }
         
-        let _month = stat[1]
-        let _year = stat[2]
+        let lastPingDate = Date(timeIntervalSince1970: TimeInterval(lastPingStat))
         
-        let lastPingDate = Date(timeIntervalSince1970: Double(stat[0]))
+        let pings = getPings(forDate: today, lastPingDate: lastPingDate)
         
-        guard let lastPingDay = DAU.calendar.components([.day], from: lastPingDate).day, let currentDay = todayComponents.day else {
-            log.error("Could not unwrap calendar components from date")
-            return nil
-        }
-         
-        let weeksMonday = Preferences.DAU.lastPingFirstMonday.value
-        // There is no lastPingFirstMondayKey preference set at first launch, meaning the week param should be set to true.
-        let isFirstLaunchWeeksMonday = weeksMonday == nil
-        let monthly = month != _month || year != _year
-        let weekly = todayComponents.weeksMonday != weeksMonday || isFirstLaunchWeeksMonday
-        // Day(of month) can stay the same if month or year changes.
-        // `monthly` parameter checks for both monthly and yearly difference.
-        // In other words, each time a monthly ping is sent, daily ping must equal true as well.
-        let daily = monthly || lastPingDay != currentDay
-        
-        log.debug("Dau stat params, daily: \(daily), weekly: \(weekly), monthly:\(monthly), lastPingDate: \(lastPingDate)")
-        if !daily && !weekly && !monthly {
-            // No changes, no ping
+        let daily = pings.first(where: { $0 == .daily })
+        let weekly = pings.first(where: { $0 == .weekly })
+        let monthly = pings.first(where: { $0 == .monthly })
+        
+        // No changes, no ping
+        if daily == nil && weekly == nil && monthly == nil {
             return nil
         }
         
-        return dauParams(daily, weekly, monthly)
+        return dauParams(daily != nil, weekly != nil, monthly != nil)
```

Comment:
I think I'd have `dauParams` just take an array too, honestly, than no need for all the logic above.